app/ante/evm/fee_checker.go

```diff
@@ -26,11 +26,6 @@ import (
 // - Tx priority is set to `effectiveGasPrice / DefaultPriorityReduction`.
 func NewDynamicFeeChecker(k DynamicFeeEVMKeeper) anteutils.TxFeeChecker {
 	return func(ctx sdk.Context, feeTx sdk.FeeTx) (sdk.Coins, int64, error) {
-		if ctx.BlockHeight() == 0 {
-			// genesis transactions: fallback to min-gas-price logic
-			return checkTxFeeWithValidatorMinGasPrices(ctx, feeTx)
-		}
-
```

Comment:
The logic here does not achieve the desired effect. In the DeliverTx stage, ctx.minGasPrice is always an empty array, Only in the CheckTx stage is ctx.minGasPrice the one we set. You can breakpoint to debug。

Reply:
Hey @luchenqun, bumping this.
Even though is true that the logic inside the `if ctx.IsCheckTx()` is not executed, the anteHandler exits in L31, without executing the subsequent logic. By removing these lines, the anteHandler will check `feeCap.LT(baseFeeInt)`. This situation breaks the `local_node.sh` and the `init-node.sh` scripts with the error message

```
gas prices too low, got: 0aevmos required: 1000000000aevmos. Please retry using a higher gas price or a higher fee: insufficient fee
```

This breaks the solidity and e2e tests.

This can be fixed either by not deleting this code or by passing the `--gas-prices "1000000000aevmos"` flag in the `evmosd gentx` transaction in both scripts

Reply:
The fundamental reason for this is that the baseFee in the genesis.json generated by **evmosd init** is 1gwei, but the gasPrice of the genesis transaction signed by **evmosd gentx** defaults to 0. So in theory we should modify the **evmosd gentx** command and pass in a reasonable gasPrice. I'll open a PR to deal with this issue when I have time. At that time, we can actually remove these lines of code.

Reply:
Also, in principle, we cannot modify the code in order to pass the e2e test. Testing is just to find logical problems in the code, don’t you think?

Reply:
After some research, I found that the **evmosd gentx** command already provides input parameters related to transaction fees, such as --gas-prices, --fees, etc. So I updated the local_node.sh and init-node.sh files. An appropriate gasPrice value is passed. If there are other places, you can also modify them. So I have removed the above code again.

Reply:
OK, thanks, please address the lint issue

Reply:
Also, this change is breaking the nix e2e tests.

Please address the issue. To run the nix e2e tests use the `make run-nix-tests`
You'll need `nix-2.13.3` to run them successfully

Reply:
> OK, thanks, please address the lint issue

done

Reply:
> Also, this change is breaking the nix e2e tests for the same reason.
> 
> Please address the issue. To run the nix e2e tests use the `make run-nix-tests` You'll need `nix-2.13.3` to run them successfully

After I installed nix (version 2.18.1), I executed **make run-nix-tests** and found that it did not end after running for nearly an hour.

<img width="1133" alt="image" src="https://github.com/evmos/evmos/assets/9688029/5892ea2b-54af-4282-a99a-dd523fb54f8c">

Reply:
<img width="1307" alt="image" src="https://github.com/evmos/evmos/assets/9688029/59296acd-b930-4a5f-b2df-ae301b134637">
I got stuck here

Reply:
I tried running **make run-nix-tests** and I did find some problem, but I don't know how to fix this.

```
github.com/evmos/evmos/v15/wallets/ledger
github.com/evmos/evmos/v15/crypto/keyring
github.com/evmos/evmos/v15/cmd/evmosd
buildPhase completed in 1 minutes 58 seconds
installing
post-installation fixup
checking for references to /private/tmp/nix-build-evmosd-latest.drv-0/ in /nix/store/xfrckkvl44jnbiasbpfiaw1vn85aggvz-evmosd-latest...
patching script interpreter paths in /nix/store/xfrckkvl44jnbiasbpfiaw1vn85aggvz-evmosd-latest
stripping (with command strip and flags -S) in  /nix/store/xfrckkvl44jnbiasbpfiaw1vn85aggvz-evmosd-latest/bin
/nix/store/xfrckkvl44jnbiasbpfiaw1vn85aggvz-evmosd-latest/bin/evmosd
pystarport init --config /Volumes/Data/Dev/go/src/github.com/evmos/evmos/tests/nix_tests/configs/rollback-test.jsonnet --data /private/tmp/pytest-of-lcq/pytest-0/rollback0 --base_port 26300 --no_remove
2023-10-31 22:26:02,118 INFO Included extra file "/private/tmp/pytest-of-lcq/pytest-0/rollback0/evmos_9000-1/tasks.ini" during parsing
2023-10-31 22:26:02,122 INFO RPC interface 'supervisor' initialized
2023-10-31 22:26:02,122 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2023-10-31 22:26:02,122 INFO supervisord started with pid 38405
2023-10-31 22:26:03,124 INFO spawned: 'evmos_9000-1-node0' with pid 38408
2023-10-31 22:26:03,125 INFO spawned: 'evmos_9000-1-node1' with pid 38409
2023-10-31 22:26:03,287 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:04,290 INFO spawned: 'evmos_9000-1-node0' with pid 38412
2023-10-31 22:26:04,472 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:04,535 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:05,538 INFO spawned: 'evmos_9000-1-node1' with pid 38414
2023-10-31 22:26:05,707 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:06,709 INFO spawned: 'evmos_9000-1-node0' with pid 38415
2023-10-31 22:26:06,871 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:07,873 INFO spawned: 'evmos_9000-1-node1' with pid 38418
2023-10-31 22:26:08,041 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:10,044 INFO spawned: 'evmos_9000-1-node0' with pid 38422
2023-10-31 22:26:10,207 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:11,208 INFO gave up: evmos_9000-1-node0 entered FATAL state, too many start retries too quickly
2023-10-31 22:26:11,210 INFO spawned: 'evmos_9000-1-node1' with pid 38423
2023-10-31 22:26:11,365 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:12,366 INFO gave up: evmos_9000-1-node1 entered FATAL state, too many start retries too quickly
FAILED2023-10-31 22:26:42,387 WARN received SIGTERM indicating exit request

test_storage_proof.py::test_basic[evmos] pystarport init --config /Volumes/Data/Dev/go/src/github.com/evmos/evmos/tests/nix_tests/configs/default.jsonnet --data /private/tmp/pytest-of-lcq/pytest-0/storage-proof0 --base_port 26800 --no_remove
2023-10-31 22:26:44,893 INFO Included extra file "/private/tmp/pytest-of-lcq/pytest-0/storage-proof0/evmos_9000-1/tasks.ini" during parsing
2023-10-31 22:26:44,895 INFO RPC interface 'supervisor' initialized
2023-10-31 22:26:44,895 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2023-10-31 22:26:44,895 INFO supervisord started with pid 38603
2023-10-31 22:26:45,899 INFO spawned: 'evmos_9000-1-node0' with pid 38607
2023-10-31 22:26:45,900 INFO spawned: 'evmos_9000-1-node1' with pid 38608
2023-10-31 22:26:46,135 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:46,135 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:47,138 INFO spawned: 'evmos_9000-1-node0' with pid 38611
2023-10-31 22:26:47,139 INFO spawned: 'evmos_9000-1-node1' with pid 38612
2023-10-31 22:26:47,340 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:47,361 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:49,363 INFO spawned: 'evmos_9000-1-node0' with pid 38619
2023-10-31 22:26:49,364 INFO spawned: 'evmos_9000-1-node1' with pid 38620
2023-10-31 22:26:49,572 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:49,588 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:52,590 INFO spawned: 'evmos_9000-1-node0' with pid 38625
2023-10-31 22:26:52,591 INFO spawned: 'evmos_9000-1-node1' with pid 38626
2023-10-31 22:26:52,805 INFO exited: evmos_9000-1-node0 (exit status 2; not expected)
2023-10-31 22:26:52,808 INFO gave up: evmos_9000-1-node0 entered FATAL state, too many start retries too quickly
2023-10-31 22:26:52,809 INFO exited: evmos_9000-1-node1 (exit status 2; not expected)
2023-10-31 22:26:53,811 INFO gave up: evmos_9000-1-node1 entered FATAL state, too many start retries too quickly

```

Reply:
The issue is related to a dependency we're using for that e2e setup. So far I couldn't find a workaround, but opened [a PR ](https://github.com/crypto-com/pystarport/pull/107)to provide the possibility to pass the `gas-prices` in the gentxs

Reply:
The relevant code has been added back

Reply:
yes, I believe will be better to avoid delaying these changes longer